Lib/glyphsLib/casting.py

```diff
@@ -53,312 +54,469 @@
-    """Prefer casting to int, but use float if necessary."""
+class RWNum(RWGlyphs):
+    """Read/write an int or float."""
 
-    val = float(string)
-    int_val = int(val)
-    return int_val if int_val == val else val
+    def read(self, src):
+        float_val = float(src)
+        int_val = int(float_val)
+        return int_val if int_val == float_val else float_val
 
+    def write(self, val):
+        assert isinstance(val, float) or isinstance(val, int)
+        return str(val)
 
-def hex_int(string):
-    """Return the hexidecimal value represented by a string."""
-    return int(string, 16)
 
+class RWHexInt(RWGlyphs):
+    """Read/write a hex int."""
 
-def vector(string, dimension):
-    """Parse a vector from a string with format {X, Y, Z, ...}."""
+    def read(self, src):
+        return int(src, 16)
 
-    rx = '{%s}' % ', '.join(['([-.e\d]+)'] * dimension)
-    return [num(i) for i in re.match(rx, string).groups()]
+    def write(self, val):
+        assert isinstance(val, int)
+        return '%04X' % val
 
 
-def point(string):
-    return vector(string, 2)
+class RWVector(RWGlyphs):
+    """Read/write a vector in curly braces."""
 
+    def __init__(self, dimension):
+        self.dimension = dimension
+        self.regex = re.compile('{%s}' % ', '.join(['([-.e\d]+)'] * dimension))
 
-def transform(string):
-    return vector(string, 6)
+    def read(self, src):
+        """Parse a vector from a string with format {X, Y, Z, ...}."""
+        return [num.read(i) for i in self.regex.match(src).groups()]
 
+    def write(self, val):
+        assert isinstance(val, list) and len(val) == self.dimension
+        return '{%s}' % (', '.join(str(v) for v in val))
 
-def node(string):
-    """Cast a node from a string with format X Y TYPE [SMOOTH]."""
 
-    rx = '([-.e\d]+) ([-.e\d]+) (LINE|CURVE|OFFCURVE|n/a)(?: (SMOOTH))?'
-    m = re.match(rx, string).groups()
-    return [num(m[0]), num(m[1]), m[2].lower(), bool(m[3])]
+class RWPoint(RWVector):
+    """Read/write a two-element vector."""
+    def __init__(self):
+        RWVector.__init__(self, 2)
 
 
-def intlist(strlist):
-    return map(int, strlist)
+class RWTransform(RWVector):
+    """Read/write a six-element vector."""
 
+    def __init__(self):
+        RWVector.__init__(self, 6)
 
-def pointlist(strlist):
-    return map(point, strlist)
 
+class RWNode(RWGlyphs):
+    """Read/write a node on an outline."""
 
-def nodelist(strlist):
-    return map(node, strlist)
+    _regex = re.compile(
+        '([-.e\d]+) ([-.e\d]+) (LINE|CURVE|OFFCURVE|n/a)(?: (SMOOTH))?')
 
+    def read(self, src):
+        """Cast a node from a string with format X Y TYPE [SMOOTH]."""
+        g = self._regex.match(src).groups()
+        return [num.read(g[0]), num.read(g[1]), g[2].lower(), bool(g[3])]
 
-def glyphs_datetime(string):
-    """Parse a datetime object from a string."""
+    def write(self, val):
+        assert isinstance(val, list) and len(val) == 4
+        v2 = val[2]
```

Comment:
Let's call this `node_type` instead of `v2`. May also want to unpack groups in read() as
`x, y, node_type, smooth = self._regex.match(src).groups()`